src/python_testing/TC_RVCOPSTATE_2_5.py

```diff
@@ -0,0 +1,273 @@
+            TestStep("3", "TH reads the SupportedModes attribute of the RVC Run Mode cluster"),
+            TestStep("4", "TH establishes a subscription to the CurrentMode attribute of the RVC Run Mode cluster of the DUT"),
+            TestStep("5", "TH sends a RVC Run Mode cluster ChangeToMode command to the DUT with NewMode set to PIXIT.CLEANMODE"),
+            TestStep("6", "Wait for DUT to leave dock and begin cleaning activities"),
+            TestStep("7", "TH reads CurrentMode attribute of the RVC Run Mode cluster"),
+            TestStep("8", "TH sends GoHome command to the DUT"),
+            TestStep("9", "Manually confirm DUT has returned to the dock and completed docking-related activities"),
+            TestStep("10", "TH reads CurrentMode attribute of the RVC Run Mode cluster"),
+        ]
+        return steps
+
+    def pics_TC_RVCOPSTATE_2_5(self) -> list[str]:
+        pics = [
+            "RVCOPSTATE.S",
+        ]
+        return pics
+
+    async def read_rvcrunm_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.RvcRunMode
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    async def read_rvcopstate_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.RvcOperationalState
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    async def read_supported_mode(self, endpoint):
+        return await self.read_rvcrunm_attribute_expect_success(endpoint=endpoint,
+                                                                attribute=Clusters.RvcRunMode.Attributes.SupportedModes)
+
+    async def read_current_mode_with_check(self, expected_mode, endpoint):
+        run_mode = await self.read_rvcrunm_attribute_expect_success(endpoint=endpoint,
+                                                                    attribute=Clusters.RvcRunMode.Attributes.CurrentMode)
+        asserts.assert_true(run_mode == expected_mode,
+                            f"Expected the current mode to be {expected_mode}, got {run_mode}")
+        return run_mode
+
+    async def read_operational_state(self, endpoint):
+        return await self.read_rvcopstate_attribute_expect_success(endpoint=endpoint,
+                                                                   attribute=Clusters.RvcOperationalState.Attributes.OperationalState)
+
+    async def read_operational_state_with_check(self, expected_operational_state):
+        operational_state = await self.read_operational_state(endpoint=self.endpoint)
+        asserts.assert_true(operational_state == expected_operational_state,
+                            # TODO:  nice conversion of states to strings
+                            f"Expected OperationalState of {expected_operational_state}, got {operational_state}"
```

Comment:
Could we use !r and get some goot result?

Reply:
https://peps.python.org/pep-0498/#s-r-and-a-are-redundant ... so either !r or repr()